File: Java/1_Array/LC_137_SingleNumberII.java
class Solution {
    public int singleNumber(int[] nums) {
        Arrays.sort(nums);
        for (int i = 0; i < nums.length - 3; i = i + 3) {
            if (nums[i] != nums[i + 2]) {
                return nums[i];
            }
        }
        return nums[nums.length - 1];
    }
}

public class LC_137_SingleNumberII {

}

File: Swift/Leetcode.playground/Sources/1_Array/LC_137_SingleNumberII.swift
class Solution {
    func singleNumber(_ nums: [Int]) -> Int {
        let sortedNums = nums.sorted()
        for i in stride(from: 0, to: nums.count - 3, by: 3) {
            if sortedNums[i] != sortedNums[i + 2] {
                return sortedNums[i]
            }
        }
        return sortedNums[nums.count - 1]
    }
}